soroban-env-host/src/builtin_contracts/stellar_asset_contract/event.rs

```diff
@@ -24,6 +44,25 @@ pub(crate) fn approve(
     Ok(())
 }
 
+pub(crate) fn transfer_maybe_with_issuer(
+    e: &Host,
+    from: Address,
+    to: Address,
+    amount: i128,
+) -> Result<(), HostError> {
+    if e.compare(&from, &to)? == Ordering::Equal {
```

Comment:
Do we prohibit both from and to being issuer?

Reply:
No we don't, so the only thing that would make sense is emitting a `transfer` in that case (or nothing at all).

Reply:
If we don't expect this to ever happen, then should this rather be an internal error?

Reply:
Well it definitely can happen at the moment. Making it an internal error wouldn't be right. We could make it a contract error, but that would be a protocol change.

Reply:
Sorry, I've misread your initial comment (for some reason I thought that we don't allow for that to happen). I think emitting no event is more appropriate, as you can't normally transfer more than your balance, but the issuer balance is always 0. So I think we should either emit no event, or make this a contract error as you've suggested. I think this is a fine change to bundle with CAP-67, as we're fixing the issuer semantics anyway.

Reply:
The issuer balance is actually `i64::MAX` (this is what `balance()` returns and was done intentionally after some discussion IIRC), and a `transfer` for over that from the issuer will fail. I still think it's fine to emit no event, but don't think it's required.

Reply:
Hmm, I see. Then I guess emitting transfer is ok, it's kind of weird semantically, but seemingly it doesn't break any invariants. Could you please add a test for that scenario?

Reply:
Yeah I added a fourth transfer in that test for this. I realized I didn't add a comment though, so I'll do that now.